unified-runtime/source/adapters/opencl/kernel.cpp

```diff
@@ -334,12 +336,18 @@ urKernelGetSubGroupInfo(ur_kernel_handle_t hKernel, ur_device_handle_t hDevice,
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urKernelRetain(ur_kernel_handle_t hKernel) {
+
+  UR_ASSERT(hKernel->getReferenceCount() > 0u, UR_RESULT_ERROR_INVALID_KERNEL);
```

Comment:
we tend to omit this kind of validation by convention, the idea being you can rely on the optional leak checking layer instead

Reply:
Sure, no problem, removed.

FWIW, I added these checks following the logic from the CUDA adapter.

Reply:
makes sense, that code probably pre-dates UR and the layers